wheat_filtration/munge.py

```diff
@@ -2,75 +2,130 @@
     Arguments:
-        source (str): the path to the txt file or directory (containing txt files)
+        source (str): the path to the text file or directory (containing text files)
             where the corpus is located. if source is a directory, it must end in /
-            or \\ (whatever is appropriate to your system)!
+            or \\ (whatever is appropriate to your system)! This function will parse all
+            files in the given directory that are not system files.
     Returns:
-        string list: a list containing strings representing documents in the corpus.
-            Documents are between 250 and 500 words. Documents end on sentences as much
-            as possible within the wordcount limits."""
+        full_corp (str): a string containing the corpus"""
 
     full_corp = ""
-    if source[-4:] == ".txt":
-        # handles case where source is a text file
-        with open(source, "r") as inf:
-            full_corp += inf.read()
-
-    else:
-        # handles case where source is directory, or throws exception if it is not
+    try:
+        with open(source, "r") as in_file:
+            full_corp += in_file.read()
+    except IsADirectoryError:
         directory = os.fsencode(source)
         for thisfile in os.listdir(directory):
             filename = os.fsdecode(thisfile)
-            if len(filename) < 4 or filename[-4:] != ".txt":
+            # skip system files
+            # maybe add functionality that allows them to give a regex for the kinds of files they want to parse
+            if filename[0] == ".":  # eg or regex not in filename
                 continue
-                # skips non text files
-            with open(source + filename, "r") as inf:
-                full_corp += inf.read()
+            try:
+                with open(source + filename, "r") as in_file:
+                    full_corp += in_file.read()
+            except IsADirectoryError:
+                continue
+    return full_corp
+
+
+def corpus_to_documents(corpus: str):
+    """Splits corpus into appropriately sized documents in the form of strings (250-500 words).
+    Arguments:
+        corpus (str): a string containing the corpus
+    Returns:
+        corpus_documents (iterable of str): a list containing strings representing documents in the corpus.
+        Documents are 250-500 words, except possibly for the last document in the corpus.
+        They end on the next sentence punctuation after 250 words, or at 500 words, whichever comes first.
+        If the sentence following a document is the last in the corpus, it is also included in the document."""
+
+    sentence_detector = nltk.data.load('tokenizers/punkt/english.pickle')
+    sentences = sentence_detector.tokenize(
+        corpus.strip())  # tokens have punc attached
+
+    corpus_documents = []
+    document_in_progress = []
+    for sentence in sentences:
+        while len(document_in_progress) > 500:
```

Comment:
250 and 500 are magic numbers right now, and should at least be global constants if not kwargs